pkg/subctl/cmd/join.go

```diff
@@ -335,7 +335,7 @@ func getNetworkDetails(config *rest.Config) *network.ClusterNetwork {
 	exitOnError("Unable to set the Kubernetes cluster connection up", err)
 	networkDetails, err := network.Discover(dynClient, clientSet)
 	if err != nil {
-		fmt.Printf("Error trying to discover network details: %s\n", err)
+		exitOnError("Error trying to discover network details: %s\n", err)
```

Comment:
This shouldn't be a catastrophic failure, this will be asked to the admin with a prompt.

Reply:
Isn't getting network details basic requirement? I think it should fail otherwise Submariner might not work

Reply:
If we can’t retrieve the network details automatically, `subctl join` will ask the administrator for the relevant info (see `getServiceCIDR` etc.). If we exit here, the program won’t get a chance to do so.

Reply:
Using warning message instead.